### PR TITLE
Handling multi-dimensional action spaces

### LINKED ISSUE #970: DDPG algorithm doesn't work with non-1d action space (probably other algs too)
### 🤖 Custom Gym Environment

```python
import gym
import numpy as np
from gym.spaces import Box

class TestEnv(gym.Env):
    metadata = {"render.modes": ["console"]}

    def __init__(self,
                max_step: int = 1,
                ) -> None:

        super().__init__()

        self.cur_step = 0
        self.max_step = max_step


        self.action_space_shape = (5,5)
        self.action_space = Box(low=-1, high=1, shape=self.action_space_shape)

        # observation space is the polarity of all normal users
        self.observation_space_shape = (5,)
        self.observation_space = Box(low=-1, high=1, shape=self.observation_space_shape)
        # self.init_obs = self.observation_space.sample()

        # self.good_counter = 0
        
        self.reset()

        return None

    def render(self, mode="console"):
        pass

    def close(self):
        pass

    def reset(self) -> np.array:
        self.cur_step = 0
        return self.observation_space.sample()

    def step(self, action: np.array):
        observation = np.array(action[0])

        self.cur_step += 1
        done = self.cur_step >= self.max_step
        info = {"info": None}
        
        reward = np.sum(action[0])

        return observation, reward, done, info

```

### Describe the bug

I expected DDPG to successfully train and return. For example, a simple example like this but using a 1d action space shape [ex: shape=np.array([25,]) works perfectly]. However, it seems like the necessary internal array reshapes haven't been implemented.

### Code example

Run this code with the above custom environment, and the error should be encountered

```python
if __name__ == "__main__":
    from stable_baselines3 import DDPG
    
    env = TestEnv(max_step=5)

    model = DDPG("MlpPolicy", env, verbose=1)
    model.learn(total_timesteps=1000, log_interval=10)
```

Here is the error and its accompanying stackframe:

```bash
Traceback (most recent call last):
  File "[FILE NAME REDACTED]", line 60, in <module>
    model.learn(total_timesteps=1000, log_interval=10)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/stable_baselines3/ddpg/ddpg.py", line 130, in learn
    return super().learn(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/stable_baselines3/td3/td3.py", line 211, in learn
    return super().learn(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 346, in learn
    rollout = self.collect_rollouts(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 595, in collect_rollouts
    self._store_transition(replay_buffer, buffer_actions, new_obs, rewards, dones, infos)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 507, in _store_transition
    replay_buffer.add(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/stable_baselines3/common/buffers.py", line 268, in add
    self.actions[self.pos] = np.array(action).copy()
ValueError: could not broadcast input array from shape (5,5) into shape (1,25)

```

### System Info
Describe the characteristic of your environment:
downloaded using pip
OS: macOS-12.2.1-arm64-arm-64bit Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000
Python: 3.10.2
Stable-Baselines3: 1.6.0
PyTorch: 1.11.0
GPU Enabled: False
Numpy: 1.22.3
Gym: 0.25.0

### Checklist

- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [x] I have checked my env using the env checker (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

closes #970

To reproduce the error:

```python
import gym
import numpy as np
from gym.spaces import Box

class TestEnv(gym.Env):
    def __init__(self) -> None:
        super().__init__()
        self._i = 0
        self.action_space = Box(low=-1, high=1, shape=(5,5))
        self.observation_space = Box(low=-1, high=1, shape=(5,))

    def render(self, mode="console"):
        pass

    def close(self):
        pass

    def reset(self) -> np.array:
        return self.observation_space.sample()

    def step(self, action: np.array):
        self._i +=1
        obs = np.zeros(5)
        done = self._i % 10 == 0
        info = {}
        reward = 0
        return obs, reward, done, info


if __name__ == "__main__":
    from stable_baselines3 import DDPG
    
    env = TestEnv()
    model = DDPG("MlpPolicy", env)
    model.learn(total_timesteps=1000)
```

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by qgallouedec** [comment]
I still need to extend this to rollout buffer and add unitest but I want to discuss the fix first

**Comment by Pulverizer2020** [comment]
The fix works on my environment! But yes, I also don't fully understnad numpy broadcasting, so I can't be sure this will always work/produce correct results.

**Comment by araffin** [comment]
@qgallouedec is it ready to review?

**Comment by qgallouedec** [comment]
Yes

**Comment by qgallouedec** [comment]
> @qgallouedec is it ready to review?

It's ready for review.

File: stable_baselines3/common/distributions.py [hunk 1]
```diff
@@ -658,7 +658,6 @@ def make_proba_distribution(
         dist_kwargs = {}
 
     if isinstance(action_space, spaces.Box):
-        assert len(action_space.shape) == 1, "Error: the action space must be a vector"
```
Comment (araffin) [inline stable_baselines3/common/distributions.py:661]: interesting, it still works when the assert is False?

**Review by araffin — COMMENTED** [review]
Thanks for the PR, LGTM, will try to test it tomorrow =) (and then probably merge)

File: stable_baselines3/common/distributions.py (hunk 1 above)
Comment (qgallouedec) [reply stable_baselines3/common/distributions.py:661]: yes, that is the point of the PR:

The distribution still works in 1D, thank to `get_action_dim`:

https://github.com/DLR-RM/stable-baselines3/blob/f6fefc3d139984c5544086ff728ca4ebfce4c0c5/stable_baselines3/common/distributions.py#L662

Then, when predicting, the distribution returns a 1D array that we need to reshape to the action shape in `BasePolicy` :

https://github.com/DLR-RM/stable-baselines3/blob/f6fefc3d139984c5544086ff728ca4ebfce4c0c5/stable_baselines3/common/policies.py#L340

and also in `ActorCriticPolicy`:

https://github.com/DLR-RM/stable-baselines3/blob/f6fefc3d139984c5544086ff728ca4ebfce4c0c5/stable_baselines3/common/policies.py#L595

**Review by araffin — APPROVED** [review]
The test you wrote was actually not checking the feature...
I added the missing lines, LGTM now, thanks =)

**Comment by qgallouedec** [comment]
Oops thanks for fixing it.

**Pull request merged**